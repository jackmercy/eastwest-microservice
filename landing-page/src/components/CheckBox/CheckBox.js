import React, { memo, useMemo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import useStyles from './checkBox.styles';

const CheckBox = ({ label, id, className, dataTest, name, ...restProps }) => {
  const classes = useStyles();
  const checkboxId = useMemo(() => id || name, []);

  return (
    <div
      className={classNames(classes.root, className)}
      data-test={dataTest || 'component-checkbox-root'}
    >
      <input
        {...restProps}
        id={checkboxId}
        name={name}
        type="checkbox"
        className={classes.input}
        data-test="component-checkbox-input"
      />
      <label
        htmlFor={checkboxId}
        className={classes.label}
        data-test="component-checkbox-label"
      >
        {label}
      </label>
    </div>
  );
};

CheckBox.propTypes = {
  label: PropTypes.node.isRequired,
  name: PropTypes.node.isRequired,
  id: PropTypes.string,
  dataTest: PropTypes.string,
  className: PropTypes.string,
};
CheckBox.defaultProps = { id: '', className: '', dataTest: '' };

export default memo(CheckBox);
