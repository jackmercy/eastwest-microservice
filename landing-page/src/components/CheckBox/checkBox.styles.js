import { createUseStyles } from 'react-jss';
import COLORS from '../../constants/COLORS';

import { hexToRgbA, rem } from '../../utils/jss';

export default createUseStyles({
  root: {
    display: 'block',
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    minWidth: rem(50),
    maxWidth: '100%',
    fontSize: rem(12),
    lineHeight: rem(16),
    color: COLORS.textGray,
    fontWeight: 500,
  },
  input: {
    margin: 0,
    padding: 0,
    height: 'initial',
    width: 'initial',
    marginBottom: 0,
    display: 'none',
    '&:checked + $label': {
      '&::before': {
        backgroundColor: COLORS.primary,
        borderColor: COLORS.primary,
      },
      '&::after': {
        content: '""',
        display: 'block',
        position: 'absolute',
        top: rem(1),
        left: rem(5),
        width: rem(3),
        height: rem(8),
        border: `solid ${COLORS.borderWhite}`,
        borderWidth: [0, rem(1.5), rem(1.5), 0],
        transform: 'rotate(45deg)',
      },
    },
    '&:disabled + $label': {
      cursor: 'default',
      color: hexToRgbA(COLORS.textGray, 0.7),
      '&::before': { cursor: 'not-allowed' },
    },
    '&:disabled:checked + $label': {
      '&::before': {
        backgroundColor: hexToRgbA(COLORS.primary, 0.7),
        borderColor: 'transparent',
      },
    },
  },
  label: {
    position: 'relative',
    cursor: 'pointer',
    color: 'inherit',
    fontSize: 'inherit',
    lineHeight: 'inherit',
    fontWeight: 'inherit',
    userSelect: 'none',
    '&::before': {
      content: '""',
      '-webkit-appearance': 'none',
      backgroundColor: 'transparent',
      border: `1px solid ${hexToRgbA(COLORS.borderBlack01, 0.2)}`,
      padding: rem(6),
      display: 'inline-block',
      position: 'relative',
      verticalAlign: 'top',
      borderRadius: rem(3),
      cursor: 'pointer',
      marginRight: rem(8),
    },
  },
});
