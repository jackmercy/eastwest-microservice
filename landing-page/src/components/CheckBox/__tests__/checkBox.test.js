import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CheckBox from '../CheckBox';

describe('Testing CheckBox component', () => {
  let wrapper;
  const props = { label: 'checkbox', name: 'checkbox' };

  beforeAll(() => {
    wrapper = mount(<CheckBox {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<CheckBox {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a label', () => {
    expect(wrapper.find('[data-test="component-checkbox-label"]')).toHaveLength(
      1,
    );
  });

  it('should have an input type=checkbox', () => {
    const selector = wrapper.find('[data-test="component-checkbox-input"]');
    expect(selector).toHaveLength(1);
    expect(selector.props().type).toEqual('checkbox');
  });
});
