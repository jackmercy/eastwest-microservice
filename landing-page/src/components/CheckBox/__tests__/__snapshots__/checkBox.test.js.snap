// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CheckBox component should render match snapshot 1`] = `
<div
  className="root-0-2-1"
  data-test="component-checkbox-root"
>
  <input
    className="input-0-2-2"
    data-test="component-checkbox-input"
    id="checkbox"
    name="checkbox"
    type="checkbox"
  />
  <label
    className="label-0-2-3"
    data-test="component-checkbox-label"
    htmlFor="checkbox"
  >
    checkbox
  </label>
</div>
`;
